vault/token_store_test.go

```diff
@@ -3643,6 +3718,201 @@ func TestTokenStore_RoleAllowedPolicies(t *testing.T) {
 	}
 }
 
+func TestTokenStore_RoleDisallowedPoliciesGlob(t *testing.T) {
+	var req *logical.Request
+	var resp *logical.Response
+	var err error
+
+	core, _, root := TestCoreUnsealed(t)
+	ts := core.tokenStore
+	ps := core.policyStore
+
+	// Create 4 different policies
```

Comment:
Do the 4 policies here need to be explicitly created, btw? I saw they are in TestTokenStore_RoleDisallowedPolicies, but not in the AllowedPolicies test, and noticed that removing these lines did not fail the test, so wondering if we can do without them ( and reduce the verbosity of the test ). I think that just hits this line https://github.com/hashicorp/vault/blob/main/vault/token_store.go#L2815 and adds some warnings in the response, but the allowed/disallowed logic gets exercised regardless, so maybe removing this code and making it similar in setup to the AllowedGlob test might not be a bad idea. wdyt ?

Reply:
I guess the only concern with not creating the policies is that while today it will still test the code effectively, in the future refactoring of the code to eg change the order of where things are checked may result in the test silently passing but not actually properly testing the code.

Happy to remove it if desired, but my gut feeling is making the test more closely reflect a real vault server setup makes it more likely to catch issues.

Reply:
I think that point totally makes sense! In that case, to keep the 2 new tests consistent with each other, should we also setup the policies `footest, testfoo, test2, test` at the start of the that?
Also, to tighten up this code a bit, maybe a helper function that takes a list of policy names and runs
```
policy, _ := ParseACLPolicy(namespace.RootNamespace, tokenCreationPolicy)
	policy.Name = "input"
	if err := ps.SetPolicy(namespace.RootContext(nil), policy); err != nil {
		t.Fatal(err)
	}
```
on each might be nice. thoughts ?